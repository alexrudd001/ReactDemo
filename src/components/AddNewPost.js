import React, { useState } from 'react';
import PostItemForm from './PostItemForm';
import ShowMessage from './ShowMessage';

const AddNewPost = () => {
    const [ messageType, setMessageType ] = useState('');
    const [loader, setLoader ] = useState(false);
    const onSubmitHandler = ({ title, body }) => {
        setMessageType('');
        setLoader(true);
        fetch('https://jsonplaceholder.typicode.com/posts', {
            method: 'POST',
            body: JSON.stringify({
                title: title,
                body: body,
                userId: Math.floor(Math.random * 100)
            }),
            headers: {
                'Content-type': 'application/json; charset=UTF-8'
            }
        })
            .then((response) => response.json())
            .then((post) => {
                setMessageType('success');
                setLoader(false);
            })
            .catch((error) => {
                setMessageType('error');
                setLoader(false);
            });
    };
    return (
        <div>
            <div className="divStyle">
                <PostItemForm onSubmit={onSubmitHandler} />
                <ShowMessage messageType={messageType} />
            </div>
            { loader && <div className="loader" /> }
        </div>
    );
};

export default AddNewPost;
