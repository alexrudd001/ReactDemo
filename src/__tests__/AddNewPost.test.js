import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddNewPost from '../components/AddNewPost';

test('renders AddNewPost correctly', () => {
    const wrapper = shallow(<AddNewPost />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

