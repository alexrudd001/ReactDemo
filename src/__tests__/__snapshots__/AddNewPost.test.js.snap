// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddNewPost correctly 1`] = `
<div>
  <div
    className="divStyle"
  >
    <PostItemForm
      onSubmit={[Function]}
    />
    <ShowMessage
      messageType=""
    />
  </div>
</div>
`;
